### PR TITLE
Provide a documentation for TypeScript FA2 API.

### PR DESCRIPTION
**Pull request by alex-kooper**

### CODE REVIEW

File: packages/fa2-interfaces/README.md [hunk 1]
```diff
@@ -139,4 +139,69 @@ type nat
 type bytes
 ```
 
-### TBD
+### Taquito Wrappers Providing Type-Safe Contract API
```
Comment (emishur) [inline packages/fa2-interfaces/README.md:142]: Update TOC to refer newly added headers

File: packages/fa2-interfaces/README.md [hunk 2]
```diff
@@ -139,4 +139,69 @@ type nat
...
+Then the wrapper can be created:
+
+```typescript
+import * as fa2 from '@oxheadalpha/fa2-interfaces';
+
+const tezosApi = fa2.tezosApi(tezos).useLambdaView(lambdaView);
+```
+
+, where `lambdaView` is an optional parameter that can specify an address of
+a "lambda view" to be used in FA2 API.
```
Comment (emishur) [inline packages/fa2-interfaces/README.md:163]: replace "lambda view" with a link to a Taquito documentation explaining what lambda views are and how to use them

File: packages/fa2-interfaces/README.md [hunk 3]
```diff
@@ -139,4 +139,69 @@ type nat
...
+
+If a custom contract and API to it is necessary, it can be implemented just
+by providing one constructor function with the following signature:
+
+```typescript
+<T>(contract: Tzip12Contract, lambdaView?: address) => T
+```
+
+The `T` type is just an object(a record of functions) and can be implemented 
+anyway possible, including using `TypeScript` `class`. In this case it has
```
Comment (emishur) [inline packages/fa2-interfaces/README.md:199]: I do not think that "Typescript" and "class" should be highlighted here

File: packages/fa2-interfaces/README.md [hunk 4]
```diff
@@ -139,4 +139,69 @@ type nat
...
+### Taquito Wrappers Providing Type-Safe Contract API
+
+`fa2-interface` package provides Taquito wrappers that allow to use type-safe API
+to FA2 contracts. To use the wrappers first an instance of Taquito toolkit
+has to be created. It can be done like this:
+
+```typescript
+import { TezosToolkit } from '@taquito/taquito';
+
+const tezos = new TezosToolkit('https://YOUR_PREFERRED_RPC_URL');
```
Comment (emishur) [inline packages/fa2-interfaces/README.md:151]: Maybe `TEZOS_NODE_RPC_URL` instead of `YOUR_PREFERRED_RPC_URL`

**Review by emishur — CHANGES_REQUESTED** [review]
- Also needs a section documenting `Fa2Contract` interface in details just as the rest of the package API
- `Fa2Contract.updateOperators()` parameters doc does not reflect recent changes

File: packages/fa2-interfaces/README.md [hunk 5]
```diff
@@ -205,3 +205,27 @@ export const MyContractApi = (
   lambdaView?: address
 ): Fa2Contract => new MyClass(contract, lambdaView)
 ```
+
+### Helpers to Run Taquito `ContractMethod`
+
+In FA2 API methods that update contract storage return `Taquito` type
```
Comment (emishur) [inline packages/fa2-interfaces/README.md:211]: `update contract storage` is not correct description.
Consider using "FA2 API methods that call/invoke contract entry points" instead

File: packages/fa2-interfaces/README.md (hunk 4 above)
Comment (alex-kooper) [reply packages/fa2-interfaces/README.md:151]: I copy & pasted it from Taquito documentation. That's what they use.

File: packages/fa2-interfaces/README.md (hunk 5 above)
Comment (alex-kooper) [reply packages/fa2-interfaces/README.md:211]: Fixed this

File: packages/fa2-interfaces/README.md (hunk 3 above)
Comment (alex-kooper) [reply packages/fa2-interfaces/README.md:199]: Unhighlighted them.

File: packages/fa2-interfaces/README.md (hunk 2 above)
Comment (alex-kooper) [reply packages/fa2-interfaces/README.md:163]: Did it, created a link.

File: packages/fa2-interfaces/README.md (hunk 1 above)
Comment (emishur) [reply packages/fa2-interfaces/README.md:142]: Still did not update the TOC

Comment (alex-kooper) [reply packages/fa2-interfaces/README.md:142]: Just did it.

File: packages/fa2-interfaces/README.md [hunk 6]
```diff
@@ -139,4 +139,146 @@ type nat
...
+```typescript
+const batch = toolkit.contract.batch();
+
+batch.withContractCall(fa2Contract.transferTokens(txs1));
+batch.withContractCall(fa2Contract.transferTokens(txs2));
+
+const op: BatchOperation = await fa2.runBatch(batch);
+```
+
+### FA2 Contract API Methods
```
Comment (emishur) [inline packages/fa2-interfaces/README.md:236]: I would follow the same pattern as the API description in the previous section.
I. e. just cut and paste a snippet of code with the API function signature and typescript doc

**Review by emishur — CHANGES_REQUESTED** [review]

File: packages/fa2-interfaces/README.md [hunk 7]
```diff
@@ -139,4 +142,146 @@ type nat
...
+
+Updates here are either add or remove commands that can be built manually or
+using batch API like this:
+
+```typescript
+const batch = operatorUpdateBatch().
+  .addOperator('tzOwner1', 'tzOperator1', 1)
+  .removeOperator('tzOwner2, 'tzOperator2', 2)
+  .addOperators([
+    { owner: 'tzOnwer3', operator: 'tzOperator3', token_id: 3 },
```
Comment (emishur) [inline packages/fa2-interfaces/README.md:280]: Misspelled "Owner" here and other places (search for "Onwer")

Comment (alex-kooper) [reply packages/fa2-interfaces/README.md]: Fixed

File: packages/fa2-interfaces/README.md (hunk 6 above)
Comment (alex-kooper) [reply packages/fa2-interfaces/README.md:236]: It is too much and becomes unreadable.

**Review by emishur — APPROVED** [review]

**Pull request merged**